src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -106,6 +107,34 @@ class SRAMInterface[T <: Data](
 
 object SRAM {
 
+  /** Generates a [[SyncReadMem]] within the current module, connected to an explicit number
+    * of read, write, and read/write ports. This SRAM abstraction has both read and write capabilities: that is,
+    * it contains at least one read accessor (a read-only or read-write port), and at least one write accessor
+    * (a write-only or read-write port).
+    *
+    * @param size The desired size of the inner `SyncReadMem`
+    * @tparam T The data type of the memory element
+    * @param numReadPorts The number of desired read ports >= 0, and (numReadPorts + numReadwritePorts) > 0
+    * @param numWritePorts The number of desired write ports >= 0, and (numWritePorts + numReadwritePorts) > 0
+    * @param numReadwritePorts The number of desired read/write ports >= 0, and the above two conditions must hold
+    * @param contents A filesystem path to a binary file to preload this SRAM's contents with
```

Comment:
Given that `loadMemoryFromFileInline` supports both Hex and Binary, why are we only exposing support for binary? I think either we should define a new type to represent the filename and file format, or we should at least add the file format as another argument to this function

Reply:
This API doc should also mention that the file path will be emitted into the Verilog (as opposed to being read during elaboration time).